packages/editor/src/components/post-author/index.js

```diff
@@ -1,49 +1,165 @@
 /**
  * WordPress dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { sprintf, __, _n } from '@wordpress/i18n';
 import { withInstanceId, compose } from '@wordpress/compose';
 import { Component } from '@wordpress/element';
 import { withSelect, withDispatch } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
+/**
+ * External dependencies
+ */
+import Autocomplete from 'accessible-autocomplete/react';
+
+import { debounce } from 'lodash';
+
 export class PostAuthor extends Component {
 	constructor() {
 		super( ...arguments );
 
 		this.setAuthorId = this.setAuthorId.bind( this );
+		this.suggestAuthor = this.suggestAuthor.bind( this );
+		this.getCurrentAuthor = this.getCurrentAuthor.bind( this );
+		this.resolveResults = this.resolveResults.bind( this );
+		this.state = {
+			postAuthor: false,
+		};
+		this.searchCache = [];
+		this.requestResults = debounce( ( query, populateResults ) => {
+			const payload = '?search=' + encodeURIComponent( query );
+			apiFetch( { path: `/wp/v2/users${ payload }` } ).then( ( results ) => {
```

Comment:
Ideally a component should never be calling `apiFetch` directly, and rather use the [core data resources](https://wordpress.org/gutenberg/handbook/designers-developers/developers/data/data-core/) as the canonical source for REST API entities. One issue with how this is implemented is we have nothing to cancel the behaviors from taking place if the component suddenly becomes unmounted, potentially leading to errors.

Reply:
Ok, I'll switch this to using core/data adding whatever support we need there. 

> we have nothing to cancel the behaviors from taking place

will core/data account for that or should i be endeavouring to cancel these requests when the component unmounts? another option would be tracking mount state and doing ending silently for request when unmounted.

Reply:
> will core/data account for that or should i be endeavouring to cancel these requests when the component unmounts?

It's a non-concern for you. In practice, it doesn't actually cancel it, but it doesn't really matter, and also helps if a future component mount also needs the same data (another reason why instance-specific fetches should be discouraged, since the data can / should be shared).

Reply:
I see we also use `apiFetch` in the user autocompleter - https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/autocompleters/user.js#L20 - i think to replace thse we'll need a new resolver, perhaps `searchAuthors`?

Reply:
@aduth Thinking about adding the capability to select both a single user by ID and to search users via the data component - currently this does not seem possible. 

Do you think I should add this to the existing `getAuthors` selector, eg the ability to pass an id or search query to the selector? or create something new?

Reply:
@aduth any feedback on this question?

Reply:
> @aduth Thinking about adding the capability to select both a single user by ID and to search users via the data component - currently this does not seem possible.
> 
> Do you think I should add this to the existing `getAuthors` selector, eg the ability to pass an id or search query to the selector? or create something new?

There's some precedent here with the [existing "entities"](https://github.com/WordPress/gutenberg/blob/535e0752051957f3e6ab7aca2e71e99ceffc9a6f/packages/core-data/src/entities.js), currently post types, taxonomies, and media items. The entities implementation will [auto-generate two selectors](https://github.com/WordPress/gutenberg/blob/535e0752051957f3e6ab7aca2e71e99ceffc9a6f/packages/core-data/src/index.js#L22-L27):

- One for retrieving a singular data entry, accepting an ID as its sole argument
- Another for querying data, accepting the REST API query as its argument

Considering `getAuthors` as a sort of partially-applied entities selector, I could see this being applied as either:

- Following the above in spirit, accepting a query object for `getAuthors` and creating a new `getAuthor` selector
- Implementing the users entity explicitly, where querying authors just becomes part of the query object passed to `getUsers`

And, in fact, it could be both, where the first is just an abstraction on the second.

I'm not sure if I've missed anything, but I'd rather not have `getAuthors` diverge as being its own special thing if it's possible for us to bring it in line with these other standard entities behaviors.